File: web/src/main/java/com/github/dkurata38/open_web_library/web/infrastructure/library/GoogleBookProperties.java
package com.github.dkurata38.open_web_library.web.infrastructure.library;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

@ConfigurationProperties(prefix = "google-book")
@ConstructorBinding
public class GoogleBookProperties {
	private final String apiKey;
	public GoogleBookProperties(String apiKey) {
		this.apiKey = apiKey;
	}

	public String getApiKey() {
		return apiKey;
	}
}

File: google-book-collation-client/src/main/kotlin/com/github/dkurata38/open_web_library/google_books/GoogleBookProperties.kt
package com.github.dkurata38.open_web_library.google_books

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConfigurationProperties(prefix = "google-book")
@ConstructorBinding
data class GoogleBookProperties(val apiKey: String) {
}
